forge-game/src/main/java/forge/game/Game.java

```diff
@@ -327,7 +326,9 @@ public Game(Iterable<RegisteredPlayer> players0, GameRules rules0, Match match0,
         int plId = 0;
         for (RegisteredPlayer psc : players0) {
             IGameEntitiesFactory factory = (IGameEntitiesFactory)psc.getPlayer();
-            Player pl = factory.createIngamePlayer(this, plId++);
+            // If the Registered Player already has a pre-assigned ID, use that. Otherwise, assign a new one.
+            Integer id = psc.getId();
+            Player pl = factory.createIngamePlayer(this, id == null ? plId++ : id);
```

Comment:
Is there a way that we have some preassigned and some assigned? We might need to update plId in those cases.

Reply:
I'll keep an eye on this, but i think in practice its ok.